src/wallet/rpcwallet.cpp

```diff
@@ -3324,6 +3477,126 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     return ValueFromAmount(nBalance);
 }
 
+UniValue z_getbalanceforaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 2)
+        throw runtime_error(
+            "z_getbalanceforaddress \"address\" ( minconf )\n"
+            "\nReturns the per-protocol (pool) balances of a Unified Address belonging to the node's wallet."
+            "\nCAUTION: If the wallet has only an incoming viewing key for this address, spends cannot be"
+            "\ndetected, and so the returned balance may be larger than the actual balance.\n"
```

Comment:
We won't immediately support incoming viewing key import anyway, so I think we can remove this caution.
```suggestion
```